tools/wptrunner/wptrunner/testrunner.py

```diff
@@ -571,12 +571,16 @@ def run_test(self):
         self.timer.start()
 
     def _timeout(self):
+        # This is executed in a different thread (threading.Timer).
         self.logger.info("Got timeout in harness")
         test = self.state.test
-        self.test_ended(test,
-                        (test.result_cls("EXTERNAL-TIMEOUT",
-                                         "TestRunner hit external timeout "
-                                         "(this may indicate a hang)"), []))
+        self.command_queue.put((
```

Comment:
Adding this to the queue seems reasonable, but I wonder if we can then end up with two `test_end` messages in the queue and should therefore downgrade the asserts in `test_ended` to be warnings in the case we get a second `test_end` message for the same test.

Reply:
I don't really understand :) The problem is that this race is inevitably a possibility if we have two threads/processses that can end the test run. But it's not a fatal error in the sense it doesn't imply bad state in the runner.

We could I suppose do better by saving the name of the previous test that ran and allowing a race here only if we get the same test_end twice in a row. But idk if that's worthwhile.

Reply:
Done.